lib/src/common/model/repos.dart

```diff
@@ -72,148 +72,147 @@ class Repository {
 
   // https clone URL
-  final String cloneUrl;
+  final String? cloneUrl;
 
-  final String sshUrl;
+  final String? sshUrl;
 
-  final String svnUrl;
+  final String? svnUrl;
 
-  final String gitUrl;
+  final String? gitUrl;
 
   /// Url to the Repository Homepage
-  final String homepage;
+  final String? homepage;
 
   /// Repository Size
-  final int size;
+  final int? size;
 
   /// Repository Stars
   @JsonKey(name: 'stargazers_count')
-  final int stargazersCount;
+  final int? stargazersCount;
 
   /// Repository Watchers
   @JsonKey(name: 'watchers_count')
-  final int watchersCount;
+  final int? watchersCount;
 
   /// Repository Language
-  final String language;
+  final String? language;
 
   /// If the Repository has Issues Enabled
   @JsonKey(name: 'has_issues')
-  final bool hasIssues;
+  final bool? hasIssues;
 
   /// If the Repository has the Wiki Enabled
   @JsonKey(name: 'has_wiki')
-  final bool hasWiki;
+  final bool? hasWiki;
 
   /// If the Repository has any Downloads
   @JsonKey(name: 'has_downloads')
-  final bool hasDownloads;
+  final bool? hasDownloads;
 
   /// If the Repository has any Github Pages
   @JsonKey(name: 'has_pages')
   final bool hasPages;
 
   /// Number of Forks
   @JsonKey(name: 'forks_count')
-  final int forksCount;
+  final int? forksCount;
 
   /// Number of Open Issues
   @JsonKey(name: 'open_issues_count')
-  final int openIssuesCount;
+  final int? openIssuesCount;
 
   /// Repository Default Branch
   @JsonKey(name: 'default_branch')
-  final String defaultBranch;
+  final String? defaultBranch;
 
   /// Number of Subscribers
   @JsonKey(name: 'subscribers_count')
-  final int subscribersCount;
+  final int? subscribersCount;
 
   /// Number of users in the network
   @JsonKey(name: 'network_count')
-  final int networkCount;
+  final int? networkCount;
 
   /// The time the repository was created at
   @JsonKey(name: 'created_at')
-  final DateTime createdAt;
+  final DateTime? createdAt;
 
   /// The last time the repository was pushed at
   @JsonKey(name: 'pushed_at')
-  final DateTime pushedAt;
+  final DateTime? pushedAt;
 
   @JsonKey(name: 'updated_at')
-  final DateTime updatedAt;
+  final DateTime? updatedAt;
 
-  final LicenseKind license;
+  final LicenseKind? license;
 
-  final bool archived;
-  final bool disabled;
+  final bool? archived;
+  final bool? disabled;
 
   factory Repository.fromJson(Map<String, dynamic> input) =>
       _$RepositoryFromJson(input);
   Map<String, dynamic> toJson() => _$RepositoryToJson(this);
 
   /// Gets the Repository Slug (Full Name).
-  RepositorySlug slug() => RepositorySlug(owner.login, name);
+  RepositorySlug slug() => RepositorySlug(owner!.login, name);
 
   @override
-  String toString() => 'Repository: ${owner.login}/$name';
+  String toString() => 'Repository: ${owner!.login}/$name';
 }
 
 @JsonSerializable(createToJson: false)
 class Tag {
-  final String name;
-  final CommitInfo commit;
+  final String? name;
```

Comment:
I haven't dug deeply but I'm curious if in practice this is ever null?

Reply:
It appears that it isn't null as far as the responses from github which are JSON deserialized into this object. This brings up another good point though .. how much should I go through the models and make as much non nullable as possible based on the github API?
What's currently in this branch is basically the minimum to meet the 2.12 null safe rules. It's very similar code as before which means things aren't likely to break since they were nullable before and there are null checks still in place.
The downside is tightening things up later would mean another major.

Reply:
Ideally we'd be more strict. If we "know" something is not going to be null, we should type it as such.

Otherwise usability is going to be miserable!

Reply:
easy for me to say, obviously. It's a lot of work to go through things and verify

Reply:
It's a balance.  I know a lot of folks are using this as an opportunity to tighten as much as they can (since it's a breaking change release already).  On the other hand you need to be practical and it's certainly better (for me anyway) to get something out there and iterate.  The workarounds for me have not been terrible but are a little awkward.  It'd be great to prevent other folks from needing to do the same dance.  (How's that for a non-answer! 😬 )

Reply:
> easy for me to say, obviously. It's a lot of work to go through things and verify

💯 

and so we need to be practical.